File: Systems-for-Design-and-Implementation/Labs/Lab8/ProiectCSharp/TransportModel/transport.model/ReservationManager.cs
using System;

namespace TransportModel.transport.model
{
    [Serializable]
    public class ReservationManager: Entity<long>
    {
        public string Name { get; set; }
        public string Password { get; set; }
    
        public ReservationManager()
        {
        }
        
        public ReservationManager(string cityName, string password)
        {
            Name = cityName;
            this.Password = password;
        }
    
        public override string ToString()
        {
            return "ReservationManager{" +
                   "CityName='" + Name + '\'' +
                   ", password='" + Password + '\'' +
                   '}';
        }
    }
}

File: S4-Systems-for-Design-and-Impl/Labs/Lab10/ProiectJava/TransportModel/src/main/java/transport/model/ReservationManager.java
package transport.model;

public class ReservationManager extends Entity<Long> {

    private String name;
    private String password;

    public ReservationManager(String name, String password) {
        this.name = name;
        this.password = password;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    @Override
    public String toString() {
        return "RM: " +
                "Name: " + name +
                "; Password: " + password;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof ReservationManager)) return false;
        ReservationManager that = (ReservationManager) o;
        return name.equals(that.name) &&
                password.equals(that.password);
    }

    @Override
    public int hashCode() {
        int result = name.hashCode();
        result = 31 * result + password.hashCode();
        return result;
    }
}
